armi/settings/caseSettings.py

```diff
@@ -368,13 +368,63 @@ def writeToYamlFile(self, fName, style="short"):
             the current state of settings
         """
         self.path = pathTools.armiAbsPath(fName)
+        if style == "medium":
+            settingsSetByUser = self.getSettingsSetByUser(self.path)
+        else:
+            settingsSetByUser = []
         with open(self.path, "w") as stream:
-            writer = self.writeToYamlStream(stream, style)
+            writer = self.writeToYamlStream(stream, style, settingsSetByUser)
         return writer
 
-    def writeToYamlStream(self, stream, style="short"):
-        """Write settings in yaml format to an arbitrary stream."""
-        writer = settingsIO.SettingsWriter(self, style=style)
+    def getSettingsSetByUser(self, fPath):
+        """
+        Grabs the list of settings in the user-defined input file so that the settings
+        can be tracked outside of a Settings Object
+
+        Parameters
+        ----------
+        fPath : str
+            The absolute file path of the settings file
+
+        Returns
+        -------
+        userSettingsNames : list
+            The settings names read in from a yaml settings file
+        """
+        from ruamel.yaml import YAML
```

Comment:
I am assuming there is some reason that this cannot be at the module level? If not, we should put it up there.

Reply:
Yea there's no reason other than me not cleaning things up properly -- done!